feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -215,49 +181,28 @@ def _execute(self, ctx, *, protocol, state_a, state_b, mapping, **inputs):
             thermo_settings=thermodynamic_settings,
             integrator_settings=integrator_settings,
             cache=ffcache,
-            has_solvent=solvent_comp is not None,
+            has_solvent=bool(solvent_comp_a),
         )
 
         # Parameterizing small molecules
         self.logger.info("Parameterizing molecules")
-        # The following creates a dictionary with all the small molecules in the states, with the structure:
-        #    Dict[SmallMoleculeComponent, openff.toolkit.Molecule]
-        # Alchemical small mols
-        alchemical_small_mols_a = {ligand_a: ligand_a.to_openff()}
-        alchemical_small_mols_b = {ligand_b: ligand_b.to_openff()}
-        all_alchemical_mols = alchemical_small_mols_a | alchemical_small_mols_b
-        # non-alchemical common small mols
-        common_small_mols = {}
-        for comp in state_a.components.values():
-            # TODO: Refactor if/when gufe provides the functionality https://github.com/OpenFreeEnergy/gufe/issues/251
-            # NOTE: This relies on gufe key for "equality", important to keep in mind
-            if (
-                isinstance(comp, SmallMoleculeComponent)
-                and comp not in all_alchemical_mols
-            ):
-                common_small_mols[comp] = comp.to_openff()
-
-        # Assign partial charges to all small mols
-        all_openff_mols = list(
-            chain(all_alchemical_mols.values(), common_small_mols.values())
-        )
-        self._assign_openff_partial_charges(
```

Comment:
If you don't assign partial charges at this point, you'll end up with bad things happening - i.e. variable partial charges everywhere being assigned with the wrong partial charge method.